### PR TITLE
[Feature] Theme mode switcher

### LINKED ISSUE #4033: Add dark mode toggle to site header
image

## ✨ Summary

[Visit the Figma file](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=11-4115&mode=design) • Contact @substrae with design questions

The goal of this ticket is to add control functionality to the newly merged dark mode changes. This ticket features updates to the styling of the existing toggle component, as well as ensuring it functions as expected. Note the temporary addition of the `BETA` label - this is intended to show users that this feature is new and that we are currently testing/collecting feedback. While we've done a significant amount of work and have been diligent about changes, I want to encourage the idea that this probably isn't going to be perfect on our first attempt and that we're open to improving the feature.

If it's helpful, and example of how the toggle should function can be found in the [Hydrogen documentation](https://hydrogen.design/en/docs/configuration/configuring-modes/#toggle).

```[tasklist]
### ✅ Requirements
- [ ] Update the toggle component to match the latest designs
- [ ] Ensure the "Preferences" selection triggers the correct mode (light/dark) based on the user's browser preference
- [ ] Ensure the "Light" selection triggers light mode regardless of the user's browser/OS preference
- [ ] Ensure the "Dark" selection triggers dark mode regardless of the user's browser/OS preference
- [ ] Ensure the selected mode is saved in local storage so that it is retained the next time the user visits the site
- [ ] Allow for a prop to control whether the "BETA" label shows in the component or not
- [ ] Design review on Dev vertical before merging
``` 

## 📸 Screenshots

<details>
  <summary>Light mode selections</summary>
  
</details> 

<details>
  <summary>Dark mode selections</summary>
  
</details> 

<details>
  <summary>Without BETA label</summary>
  
</details> 

## ❓ Questions
- [ ] Does this require a new context provider?

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #4033 

## 👋 Introduction

Adds the theme switcher to the header with the beta label.

## 🕵️ Details

This also includes some changes to the base `ToggleGroup` component, removing the option to change the colour.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

> [!TIP]
> If you don't want to change your system settings and just emulate it, here is how you do that you can follow [this guide](https://devtoolstips.org/tips/en/emulate-color-schemes/) for each browser.

1. Build `npm run dev`
2. Navigate to home page
3. Confirm the switcher appears and matches [the mockup](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=11-4115&mode=design&t=PUAWqgbAuvjPXtIT-0)
4. Confirm it starts of the first option
5. Change modes confirming it updates the theme
6. Refresh the page confirming your settings persist
7. Change back to the first option (preference)
8. Change your `prefers-color-scheme`
9. Confirm the page updates to reflect your settings
10. Repeat steps 3-9 on the IAP

## 📸 Screenshot

Screenshot 2024-02-06 123757
Screenshot 2024-02-06 123801
Screenshot 2024-02-06 123805
Screenshot 2024-02-06 123809

### CODE REVIEW

**Review by substrae — APPROVED** [review]
Looks great! Awesome work @esizer. The toggles all work, choice is retained across sessions, and it's respecting my browser choice as well.

File: packages/theme/src/components/ThemeProvider.tsx [hunk 1]
```diff
@@ -13,7 +13,8 @@ import {
 } from "../types";
 
 export interface ThemeState {
-  mode: ThemeMode;
+  fullMode: ThemeMode;
+  mode: Omit<ThemeMode, "pref">;
```
Comment (petertgiles) [inline packages/theme/src/components/ThemeProvider.tsx:17]: Could you please add some comments to this interface?  It's not obvious to me what fullMode vs mode vs key is.

File: packages/ui/src/components/ToggleGroup/ToggleGroup.tsx [hunk 2]
```diff
@@ -4,106 +4,65 @@
...
-  },
-};
-
 const Item = React.forwardRef<
   React.ElementRef<typeof ToggleGroupPrimitive.Item>,
   React.ComponentPropsWithoutRef<typeof ToggleGroupPrimitive.Item>
 >((props, forwardedRef) => (
   <ToggleGroupPrimitive.Item
     data-h2-align-items="base(center)"
-    data-h2-background-color="
```
Comment (petertgiles) [inline packages/ui/src/components/ToggleGroup/ToggleGroup.tsx:66]: We don't need the colours anymore?

File: packages/theme/src/components/ThemeProvider.tsx [hunk 3]
```diff
@@ -67,6 +69,12 @@ const ThemeProvider = ({
     getDefaultTheme(override),
   );
   const { key, mode } = theme;
+  let computedMode: ThemeMode = mode;
```
Comment (petertgiles) [inline packages/theme/src/components/ThemeProvider.tsx:72]: I think that if this is truly beta then the default mode on L45 should be light, not pref.  The user should have to opt into it, I think.

Unfortunately, all our existing users proably already have "pref" in their localStorage.

**Review by petertgiles — DISMISSED** [review]
Nice!  :sunglasses: I flagged a few items to consider but nothing blocking.

File: packages/theme/src/components/ThemeProvider.tsx (hunk 1 above)
Comment (esizer) [reply packages/theme/src/components/ThemeProvider.tsx:22]: Yeah, that is a good idea! Added in [add documentation to different mode values](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9217/commits/c979e407889d7c6008359cff5c149ebe0bf01c4c). Hopefully that makes sense 😅

**Comment by petertgiles** [comment]
Works great!  I added a deployment tag to the issue.  Please add a little blurb about the new flag to the PR.

**Comment by petertgiles** [comment]
Created  ♻️ Remove flag - FEATURE_DARK_MODE #9291

File: packages/theme/src/components/ThemeProvider.tsx (hunk 3 above)
Comment (esizer) [reply packages/theme/src/components/ThemeProvider.tsx:79]: Spoke with @substrae and the fact that we likely have already stored "pref" and the beta flag is just us being super careful then we should be good to merge as is.

Users can always set light mode if they don't like our dark mode too.. I guess 🤷‍♀️ 😅

**Comment by brindasasi** [comment]
> Created ♻️ Remove flag - FEATURE_DARK_MODE #9291

woh .. Almost missed this.. no instructions on this to create this flag

**Comment by esizer** [comment]
> woh .. Almost missed this.. no instructions on this to create this flag

Oops, sorry Brinda! Thanks for covering up for me 😅

**Pull request merged**